### ISSUE TITLE
When I try inactive a language not change default values

### ISSUE THREAD

**Issue by kurkop** [odoo/odoo#15979]
Impacted versions:
9.0, 10.0

Steps to reproduce:

Create a new database with language English by default and add a new language. Go to settings->translate->languages (in debug mode), inactive English. Go to contacts and "Create". 

Related: https://github.com/odoo/odoo/issues/3854

Current behavior:

Frontend error: Try read en_US language and add to partner, but this not are avalaible.

Expected behavior:
Change default language for all partners.

I need change the default value manually for create new partners:
```
UPDATE public.ir_values
   SET value='S''es_CO''
p1
.'
 WHERE id=1;
```

Video/Screenshot link (optional):

**Comment by Yenthe666** [comment]
@mart-e could you give input?

**Comment by mart-e** [comment]
Hello,

What you forgot to specify in your report is that you need also to change all your users language, including the inactive ones (like public user).

![selection_026](https://cloud.githubusercontent.com/assets/564822/24187654/3ba68e78-0ede-11e7-870f-9cb5939bdbcb.png)

Even if inactive, I am not sure to understand where do you get an error.
Could you copy the error message?

**Comment by kurkop** [comment]
Hi @mart-e, excuse me. I change before language for all users. The error appears in:
Click in Contacts menu, click in create button. 
```
  File "/home/src/odoo-cc/odoo/models.py", line 4892, in _convert_to_cache
    for name, value in values.iteritems()
  File "/home/src/odoo-cc/odoo/models.py", line 4893, in <dictcomp>
    if name in fields
  File "/home/src/odoo-cc/odoo/fields.py", line 1738, in convert_to_cache
    raise ValueError("Wrong value for %s: %r" % (self, value))
ValueError: Wrong value for res.partner.lang: u'en_US'
```
I trace error and I can see that the values contains 'lang': u'en_US', because in ir.values has this value (id=1).

**Comment by blaggacao** [comment]
Transliterpreting: "We traced the issue to the point, that when uninstalling the language, it does not 'clean' - or better - 'update' `ir.values`-default-value nor does it update `res.partner`". Constraint checks don't apply there neither.

**Comment by mart-e** [comment]
Changing/checking the ir.value in a generic way is not realistic with the current model (I made an attempt at #16009 but without success) without performance degradation (we should be careful not to slow down massive CSV import for instance).

In the end, we made a special case for res.lang to remove the default value if somebody deactivated the language. The scenario is too specific to be worth slowing down all default_get.
